src/book/02-system/01-foundations/07-director.mdx

```diff
@@ -0,0 +1,447 @@
+
+#### Start
+
+A Start Provider is used to set up the state of a Provider when it starts running. Start is executed when a Provider group has no active Tasks and then is given to one, ie it gains control.
+
+For example, the walk may need a Start Provider to initialise the state of the walk engine before it starts running.
+
+```cpp
+on<Start<Walk>>().then( { // this is a Start Provider type for the Walk Task
+    // Initialise walk
+    delta = NUClear::clock::now();
+    walk_engine.reset();
+    current_orders = Eigen::Vector3d::Zero();
+});
+```
+
+#### Provide
+
+Provide is the normal Provider which executes the functionality for a Task, as described previously. It will run after Start, if a Start exists.
+
+```cpp
+on<Provide<Walk>>().then( { // a Provide Provider type for the Walk Task
+    // Run the walk
+    auto goals = calculate_joint_goals();
+    emit<Task>(std::make_unique<LegIK>(goals));
+});
+```
+
+#### Stop
+
+Stop Providers run when a Provider group is losing control and will no longer run. That is, when the Provider group has no active Task. This is useful for performing a cleanup of the Provider.
+
+A Stop Provider cannot run unless either a Start or Provide has run for that Provider group.
+
+For example:
+
+```cpp
+on<Stop<Walk>>().then({ // a Stop Provider type for the Walk Task
+    // Log that the walk has stopped
+    log<NUClear::TRACE>("Walk Stopped");
+});
+```
+
+### Needs
+
+The Needs DSL word is used to ensure a Provider can take control of subtasks, otherwise it cannot run.
+
+```cpp
+on<Provide<Walk>, Needs<LegIK>>().then(...);
+```
+
+In this example, the Walk Provider will only run if it can take over the LegIK Task.
+
+If another, higher priority, Provider has already taken over the LegIK Task, then this Provider will not run. If the LegIK is not being used by any other Provider, or it has higher priority than the Provider controlling the LegIK, then this Provider will take over the LegIK Task.
+
+Without `Needs`, the Provider will still run and can emit subtasks, but the subtasks will not do anything if another Provider has access to them.
+
+### When
+
+A When condition is used to ensure that a Provider will only execute when the specified condition is met.
+These conditions are provided using a global enum message along with a statement that describes what condition you want.
+For example, if you wanted a kick engine to only run if the robot is statically stable or better then you would write:
+
+```cpp
+on<Provide<Kick>, When<Stability, std::greater_equal, Stability::STANDING>>().then(...);
+```
+
+These conditions are updated by using normal NUClear emits. For example if
+
+```cpp
+emit(std::make_unique<Stability>(Stability::STATIC));
+```
+
+was emitted in another reaction, then the `When` condition would be met and the Provider would run.
```

Comment:
This becomes a code block so either the second line should be capitalised or it should be an inline code snippet. I think the latter flows better but up to you

```suggestion
Was emitted in another reaction, then the `When` condition would be met and the Provider would run.
```